legal-api/src/legal_api/services/authz.py

```diff
@@ -537,6 +555,65 @@ def get_allowable_actions(jwt: JwtManager, business: Business):
     return result
 
 
+def get_could_file(business: Business,
```

Comment:
If I compare this function with `get_allowed_filings` the only change you have done here is to remove `business_blocker_check`. can you explain why?

Reply:
So the intent behind could_file is to have an api answer on whether a user could ever do a type of filing. So while allowedActions says you can/can't do this for this business, couldFile responds that you could do it for a business of this legalType. We intend to use it to show/hide options, IE couldFile --> visibility, allowedActions --> disabled/enabled

Reply:
I think there "could" (pun intended) be other things that determiend whether or not to show/hide an action (eg, current user).

I'd prefer to defer this implementation until a complete list of show/hide conditions is known.

Reply:
The current user and the business type is considered

Reply:
If you get all the filings a user can file for a business, how do you check if that action is blocked or not?

Reply:
We still use allowedActions and disable/enable the button based on that flag which checks blocked. the Blocked flag is business specific and doesn't tell you whether or not the current user could file a certain type of filing for a different business of the same type. IE they could file this for a BC business (so it's in could file) but they aren't allowed to for this business (so it isn't in allowedActions)

Reply:
well, if i'm understanding things correctly.  it doesn't sound like the functionality you are looking for fits under the GET /businesses/<identifier> endpoint.  

It sounds like you need another endpoint that retrieves allowed filings based off of current user and business type.  

So I don't think it makes sense that you are trying to retrieve this info from a specific business?

Reply:
> The current user and the business type is considered

The current implementation is simplistic (basically, on or off depending on product key). I know that some user types will have more or less permissions than others (ie, staff > Service BC > reg user).

Before implementing get_could_files, I'd like to see what conditions (in the old UI) are used for showing/hiding actions, and making sure they're considered here.

Reply:
I guess, in a nutshell, I'd like to see this feature generalized.

Reply:
Obsolete comment. Resolving.